x/stake/keeper/delegation.go

```diff
@@ -677,7 +696,9 @@ func (k Keeper) BeginUnbonding(ctx sdk.Context,
 		CrossStake:     ctx.CrossStake(),
 	}
 	k.SetUnbondingDelegation(ctx, ubd)
```

Comment:
I think we dont need to `SetUnbondingDelegation`  after the first hardfork, otherwise there is no chance to delete it , and block the following undelegation, because of the code:
```
	_, found := k.GetUnbondingDelegation(ctx, delAddr, valAddr)
	if found {
		return types.UnbondingDelegation{}, types.ErrExistingUnbondingDelegation(k.Codespace())
	}
```